Support setting safeToCompareToEmptyOrDeleted in MemberBasedHash

anttijk opened the pull request:
#### acb1c76129d479ba1b60f43ab9d4c446205dc957
<pre>
Support setting safeToCompareToEmptyOrDeleted in MemberBasedHash
<a href="https://bugs.webkit.org/show_bug.cgi?id=301816">https://bugs.webkit.org/show_bug.cgi?id=301816</a>
<a href="https://rdar.apple.com/163877354">rdar://163877354</a>

Reviewed by Darin Adler and Sam Weinig.

Allow Hash types with safeToCompareToEmptyOrDeleted = true to be replaced with MemberBasedHash too.

To signal that the comparison is safe the type itself can have optional

static constexpr bool safeToCompareToHashTableEmptyOrDeletedValue = true;

field which is then picked up by MemberBasedHash.

Use this to remove lots of boilerplate.
Also clean up a few additional cases with safeToCompareToEmptyOrDeleted = false.

* Source/JavaScriptCore/b3/B3CheckSpecial.h:
(JSC::B3::CheckSpecialKeyHash::hash): Deleted.
(JSC::B3::CheckSpecialKeyHash::equal): Deleted.
* Source/JavaScriptCore/b3/B3Kind.h:
(JSC::B3::KindHash::hash): Deleted.
(JSC::B3::KindHash::equal): Deleted.
* Source/JavaScriptCore/b3/B3ValueKey.h:
(JSC::B3::ValueKeyHash::hash): Deleted.
(JSC::B3::ValueKeyHash::equal): Deleted.
* Source/JavaScriptCore/b3/air/AirArg.h:
(JSC::B3::Air::ArgHash::hash): Deleted.
(JSC::B3::Air::ArgHash::equal): Deleted.
* Source/JavaScriptCore/b3/air/AirCode.h:
* Source/JavaScriptCore/b3/air/AirTmp.h:
(JSC::B3::Air::TmpHash::hash): Deleted.
(JSC::B3::Air::TmpHash::equal): Deleted.
* Source/JavaScriptCore/bytecode/BytecodeIndex.h:
(JSC::BytecodeIndexHash::hash): Deleted.
(JSC::BytecodeIndexHash::equal): Deleted.
* Source/JavaScriptCore/bytecode/CallVariant.h:
(JSC::CallVariantHash::hash): Deleted.
(JSC::CallVariantHash::equal): Deleted.
* Source/JavaScriptCore/bytecode/CodeOrigin.h:
(JSC::CodeOriginHash::hash): Deleted.
(JSC::CodeOriginHash::equal): Deleted.
* Source/JavaScriptCore/bytecode/DFGExitProfile.h:
(JSC::DFG::FrequentExitSiteHash::hash): Deleted.
(JSC::DFG::FrequentExitSiteHash::equal): Deleted.
* Source/JavaScriptCore/bytecode/DirectEvalCodeCache.h:
(JSC::DirectEvalCodeCache::CacheKey::Hash::hash): Deleted.
(JSC::DirectEvalCodeCache::CacheKey::Hash::equal): Deleted.
* Source/JavaScriptCore/bytecode/LazyOperandValueProfile.h:
(JSC::LazyOperandValueProfileKeyHash::hash): Deleted.
(JSC::LazyOperandValueProfileKeyHash::equal): Deleted.
* Source/JavaScriptCore/bytecode/ObjectPropertyCondition.h:
(JSC::ObjectPropertyConditionHash::hash): Deleted.
(JSC::ObjectPropertyConditionHash::equal): Deleted.
* Source/JavaScriptCore/bytecode/PropertyCondition.h:
(JSC::PropertyConditionHash::hash): Deleted.
(JSC::PropertyConditionHash::equal): Deleted.
* Source/JavaScriptCore/dfg/DFGAbstractHeap.h:
(JSC::DFG::AbstractHeapHash::hash): Deleted.
(JSC::DFG::AbstractHeapHash::equal): Deleted.
* Source/JavaScriptCore/dfg/DFGDesiredGlobalProperty.h:
(JSC::DFG::DesiredGlobalPropertyHash::hash): Deleted.
(JSC::DFG::DesiredGlobalPropertyHash::equal): Deleted.
* Source/JavaScriptCore/dfg/DFGHeapLocation.h:
(JSC::DFG::HeapLocationHash::hash): Deleted.
(JSC::DFG::HeapLocationHash::equal): Deleted.
* Source/JavaScriptCore/dfg/DFGMinifiedID.h:
(JSC::DFG::MinifiedIDHash::hash): Deleted.
(JSC::DFG::MinifiedIDHash::equal): Deleted.
* Source/JavaScriptCore/dfg/DFGNodeFlowProjection.h:
(JSC::DFG::NodeFlowProjectionHash::hash): Deleted.
(JSC::DFG::NodeFlowProjectionHash::equal): Deleted.
* Source/JavaScriptCore/dfg/DFGPromotedHeapLocation.h:
(JSC::DFG::PromotedLocationDescriptorHash::hash): Deleted.
(JSC::DFG::PromotedLocationDescriptorHash::equal): Deleted.
(JSC::DFG::PromotedHeapLocationHash::hash): Deleted.
(JSC::DFG::PromotedHeapLocationHash::equal): Deleted.
* Source/JavaScriptCore/dfg/DFGPropertyTypeKey.h:
(JSC::DFG::PropertyTypeKeyHash::hash): Deleted.
(JSC::DFG::PropertyTypeKeyHash::equal): Deleted.
* Source/JavaScriptCore/dfg/DFGPureValue.h:
(JSC::DFG::PureValueHash::hash): Deleted.
(JSC::DFG::PureValueHash::equal): Deleted.
* Source/JavaScriptCore/ftl/FTLLocation.h:
(JSC::FTL::LocationHash::hash): Deleted.
(JSC::FTL::LocationHash::equal): Deleted.
* Source/JavaScriptCore/interpreter/CallFrame.h:
(JSC::CallSiteIndexHash::hash): Deleted.
(JSC::CallSiteIndexHash::equal): Deleted.
* Source/JavaScriptCore/jit/ICStats.h:
(JSC::ICEventHash::hash): Deleted.
(JSC::ICEventHash::equal): Deleted.
* Source/JavaScriptCore/jit/JITCompilationKey.h:
(JSC::JITCompilationKeyHash::hash): Deleted.
(JSC::JITCompilationKeyHash::equal): Deleted.
* Source/JavaScriptCore/jit/Reg.h:
(JSC::RegHash::hash): Deleted.
(JSC::RegHash::equal): Deleted.
* Source/JavaScriptCore/parser/SourceCodeKey.h:
(JSC::SourceCodeKey::Hash::hash): Deleted.
(JSC::SourceCodeKey::Hash::equal): Deleted.
* Source/JavaScriptCore/profiler/ProfilerOrigin.h:
(JSC::Profiler::OriginHash::hash): Deleted.
(JSC::Profiler::OriginHash::equal): Deleted.
* Source/JavaScriptCore/profiler/ProfilerOriginStack.h:
(JSC::Profiler::OriginStackHash::hash): Deleted.
(JSC::Profiler::OriginStackHash::equal): Deleted.
* Source/JavaScriptCore/runtime/CodeCache.h:
* Source/JavaScriptCore/runtime/ControlFlowProfiler.h:
(JSC::BasicBlockKeyHash::hash): Deleted.
(JSC::BasicBlockKeyHash::equal): Deleted.
* Source/JavaScriptCore/runtime/PrototypeKey.h:
(JSC::PrototypeKeyHash::hash): Deleted.
(JSC::PrototypeKeyHash::equal): Deleted.
* Source/JavaScriptCore/runtime/TypeProfiler.h:
(JSC::QueryKeyHash::hash): Deleted.
(JSC::QueryKeyHash::equal): Deleted.
* Source/JavaScriptCore/runtime/VarOffset.h:
(JSC::VarOffsetHash::hash): Deleted.
(JSC::VarOffsetHash::equal): Deleted.
* Source/WTF/wtf/GenericHashKey.h:
* Source/WTF/wtf/HashFunctions.h:
* Source/WebCore/platform/Cookie.h:
(WebCore::CookieHash::hash): Deleted.
(WebCore::CookieHash::equal): Deleted.
* Source/WebCore/platform/Site.h:
(WebCore::Site::Hash::hash): Deleted.
(WebCore::Site::Hash::equal): Deleted.
* Source/WebCore/platform/graphics/FontCache.cpp:
(WebCore::FontDataCacheKeyHash::hash): Deleted.
(WebCore::FontDataCacheKeyHash::equal): Deleted.
* Source/WebCore/platform/graphics/FontPlatformData.h:
* Source/WebCore/platform/graphics/WidthCache.h:
(WebCore::WidthCache::SmallStringKeyHash::hash): Deleted.
(WebCore::WidthCache::SmallStringKeyHash::equal): Deleted.
* Source/WebCore/rendering/CSSValueKey.h:
(WTF::CSSValueKeyHash::hash): Deleted.
(WTF::CSSValueKeyHash::equal): Deleted.
* Source/WebCore/rendering/TextAutoSizing.h:
(WebCore::TextAutoSizingHash::hash): Deleted.
(WebCore::TextAutoSizingHash::equal): Deleted.
* Source/WebKit/Shared/WebFoundTextRange.h:
(WTF::WebFoundTextRangePDFDataHash::hash): Deleted.
(WTF::WebFoundTextRangePDFDataHash::equal): Deleted.
(WTF::WebFoundTextRangeHash::hash): Deleted.
(WTF::WebFoundTextRangeHash::equal): Deleted.
* Source/WebKit/UIProcess/Extensions/WebExtensionMatchPattern.h:
(WTF::WebExtensionMatchPatternHash::hash): Deleted.
(WTF::WebExtensionMatchPatternHash::equal): Deleted.

Canonical link: <a href="https://commits.webkit.org/302472@main">https://commits.webkit.org/302472@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d8bb5fb8a70b8eaabcbbd2fb17e92652b30ddbfc

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/129198 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/1456 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/40034 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/136575 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/80590 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/7c16b988-2a33-4537-be21-9b16f9af9c1f) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/131069 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/1388 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/1333 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/98382 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/66282 "Passed tests") | [❌ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/a266576d-165c-4b38-b51c-d4d52d1ed107) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/132145 "Passed tests") | [⏳ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/iOS-26-Simulator-WK2-Tests-EWS "Waiting to run tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/115735 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/79031 "Passed tests") | | ⏳ 🛠 vision-apple 
| | [⏳ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/iOS-26-Simulator-WPT-WK2-Tests-EWS "Waiting to run tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/33853 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/79854 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/121189 "Built successfully and passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/109456 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/34348 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/139049 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/127650 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/1248 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/1207 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/106913 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/1299 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/112071 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/106749 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27177 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/1031 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/30595 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/53848 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/1321 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/64675 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/160664 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/1146 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/40105 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/1191 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/1245 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WTF/wtf/HashFunctions.h:
```diff
@@ -266,7 +266,12 @@ namespace WTF {
         static bool equal(const T& a, const T& b) { return a == b; }
-        static constexpr bool safeToCompareToEmptyOrDeleted = false;
+        static constexpr bool safeToCompareToEmptyOrDeleted = [] {
+            if constexpr (requires { T::safeToCompareToHashTableEmptyOrDeletedValue; })
```
Is there a way to write this that would fail to compile (rather than just returning either `true` or `false`) if this is a static function member or static data member of a type other than `bool`? I can't think of one but maybe there is a way.

weinig replied:
You could probably do something like:

```
if constexpr (requires { T::safeToCompareToHashTableEmptyOrDeletedValue; }) {
    static_assert(std::same_as<decltype(T::safeToCompareToHashTableEmptyOrDeletedValue), bool>);
    return T::safeToCompareToHashTableEmptyOrDeletedValue;
} else
    return false 
```

anttijk replied:
`static_assert(std::same_as<decltype(T::safeToCompareToHashTableEmptyOrDeletedValue), const bool>);`

anttijk committed:
Support setting safeToCompareToEmptyOrDeleted in MemberBasedHash
https://bugs.webkit.org/show_bug.cgi?id=301816
rdar://163877354

Reviewed by Darin Adler and Sam Weinig.

Allow Hash types with safeToCompareToEmptyOrDeleted = true to be replaced with MemberBasedHash too.

To signal that the comparison is safe the type itself can have optional

static constexpr bool safeToCompareToHashTableEmptyOrDeletedValue = true;

field which is then picked up by MemberBasedHash.

Use this to remove lots of boilerplate.
Also clean up a few additional cases with safeToCompareToEmptyOrDeleted = false.

* Source/JavaScriptCore/b3/B3CheckSpecial.h:
(JSC::B3::CheckSpecialKeyHash::hash): Deleted.
(JSC::B3::CheckSpecialKeyHash::equal): Deleted.
* Source/JavaScriptCore/b3/B3Kind.h:
(JSC::B3::KindHash::hash): Deleted.
(JSC::B3::KindHash::equal): Deleted.
* Source/JavaScriptCore/b3/B3ValueKey.h:
(JSC::B3::ValueKeyHash::hash): Deleted.
(JSC::B3::ValueKeyHash::equal): Deleted.
* Source/JavaScriptCore/b3/air/AirArg.h:
(JSC::B3::Air::ArgHash::hash): Deleted.
(JSC::B3::Air::ArgHash::equal): Deleted.
* Source/JavaScriptCore/b3/air/AirCode.h:
* Source/JavaScriptCore/b3/air/AirTmp.h:
(JSC::B3::Air::TmpHash::hash): Deleted.
(JSC::B3::Air::TmpHash::equal): Deleted.
* Source/JavaScriptCore/bytecode/BytecodeIndex.h:
(JSC::BytecodeIndexHash::hash): Deleted.
(JSC::BytecodeIndexHash::equal): Deleted.
* Source/JavaScriptCore/bytecode/CallVariant.h:
(JSC::CallVariantHash::hash): Deleted.
(JSC::CallVariantHash::equal): Deleted.
* Source/JavaScriptCore/bytecode/CodeOrigin.h:
(JSC::CodeOriginHash::hash): Deleted.
(JSC::CodeOriginHash::equal): Deleted.
* Source/JavaScriptCore/bytecode/DFGExitProfile.h:
(JSC::DFG::FrequentExitSiteHash::hash): Deleted.
(JSC::DFG::FrequentExitSiteHash::equal): Deleted.
* Source/JavaScriptCore/bytecode/DirectEvalCodeCache.h:
(JSC::DirectEvalCodeCache::CacheKey::Hash::hash): Deleted.
(JSC::DirectEvalCodeCache::CacheKey::Hash::equal): Deleted.
* Source/JavaScriptCore/bytecode/LazyOperandValueProfile.h:
(JSC::LazyOperandValueProfileKeyHash::hash): Deleted.
(JSC::LazyOperandValueProfileKeyHash::equal): Deleted.
* Source/JavaScriptCore/bytecode/ObjectPropertyCondition.h:
(JSC::ObjectPropertyConditionHash::hash): Deleted.
(JSC::ObjectPropertyConditionHash::equal): Deleted.
* Source/JavaScriptCore/bytecode/PropertyCondition.h:
(JSC::PropertyConditionHash::hash): Deleted.
(JSC::PropertyConditionHash::equal): Deleted.
* Source/JavaScriptCore/dfg/DFGAbstractHeap.h:
(JSC::DFG::AbstractHeapHash::hash): Deleted.
(JSC::DFG::AbstractHeapHash::equal): Deleted.
* Source/JavaScriptCore/dfg/DFGDesiredGlobalProperty.h:
(JSC::DFG::DesiredGlobalPropertyHash::hash): Deleted.
(JSC::DFG::DesiredGlobalPropertyHash::equal): Deleted.
* Source/JavaScriptCore/dfg/DFGHeapLocation.h:
(JSC::DFG::HeapLocationHash::hash): Deleted.
(JSC::DFG::HeapLocationHash::equal): Deleted.
* Source/JavaScriptCore/dfg/DFGMinifiedID.h:
(JSC::DFG::MinifiedIDHash::hash): Deleted.
(JSC::DFG::MinifiedIDHash::equal): Deleted.
* Source/JavaScriptCore/dfg/DFGNodeFlowProjection.h:
(JSC::DFG::NodeFlowProjectionHash::hash): Deleted.
(JSC::DFG::NodeFlowProjectionHash::equal): Deleted.
* Source/JavaScriptCore/dfg/DFGPromotedHeapLocation.h:
(JSC::DFG::PromotedLocationDescriptorHash::hash): Deleted.
(JSC::DFG::PromotedLocationDescriptorHash::equal): Deleted.
(JSC::DFG::PromotedHeapLocationHash::hash): Deleted.
(JSC::DFG::PromotedHeapLocationHash::equal): Deleted.
* Source/JavaScriptCore/dfg/DFGPropertyTypeKey.h:
(JSC::DFG::PropertyTypeKeyHash::hash): Deleted.
(JSC::DFG::PropertyTypeKeyHash::equal): Deleted.
* Source/JavaScriptCore/dfg/DFGPureValue.h:
(JSC::DFG::PureValueHash::hash): Deleted.
(JSC::DFG::PureValueHash::equal): Deleted.
* Source/JavaScriptCore/ftl/FTLLocation.h:
(JSC::FTL::LocationHash::hash): Deleted.
(JSC::FTL::LocationHash::equal): Deleted.
* Source/JavaScriptCore/interpreter/CallFrame.h:
(JSC::CallSiteIndexHash::hash): Deleted.
(JSC::CallSiteIndexHash::equal): Deleted.
* Source/JavaScriptCore/jit/ICStats.h:
(JSC::ICEventHash::hash): Deleted.
(JSC::ICEventHash::equal): Deleted.
* Source/JavaScriptCore/jit/JITCompilationKey.h:
(JSC::JITCompilationKeyHash::hash): Deleted.
(JSC::JITCompilationKeyHash::equal): Deleted.
* Source/JavaScriptCore/jit/Reg.h:
(JSC::RegHash::hash): Deleted.
(JSC::RegHash::equal): Deleted.
* Source/JavaScriptCore/parser/SourceCodeKey.h:
(JSC::SourceCodeKey::Hash::hash): Deleted.
(JSC::SourceCodeKey::Hash::equal): Deleted.
* Source/JavaScriptCore/profiler/ProfilerOrigin.h:
(JSC::Profiler::OriginHash::hash): Deleted.
(JSC::Profiler::OriginHash::equal): Deleted.
* Source/JavaScriptCore/profiler/ProfilerOriginStack.h:
(JSC::Profiler::OriginStackHash::hash): Deleted.
(JSC::Profiler::OriginStackHash::equal): Deleted.
* Source/JavaScriptCore/runtime/CodeCache.h:
* Source/JavaScriptCore/runtime/ControlFlowProfiler.h:
(JSC::BasicBlockKeyHash::hash): Deleted.
(JSC::BasicBlockKeyHash::equal): Deleted.
* Source/JavaScriptCore/runtime/PrototypeKey.h:
(JSC::PrototypeKeyHash::hash): Deleted.
(JSC::PrototypeKeyHash::equal): Deleted.
* Source/JavaScriptCore/runtime/TypeProfiler.h:
(JSC::QueryKeyHash::hash): Deleted.
(JSC::QueryKeyHash::equal): Deleted.
* Source/JavaScriptCore/runtime/VarOffset.h:
(JSC::VarOffsetHash::hash): Deleted.
(JSC::VarOffsetHash::equal): Deleted.
* Source/WTF/wtf/GenericHashKey.h:
* Source/WTF/wtf/HashFunctions.h:
* Source/WebCore/platform/Cookie.h:
(WebCore::CookieHash::hash): Deleted.
(WebCore::CookieHash::equal): Deleted.
* Source/WebCore/platform/Site.h:
(WebCore::Site::Hash::hash): Deleted.
(WebCore::Site::Hash::equal): Deleted.
* Source/WebCore/platform/graphics/FontCache.cpp:
(WebCore::FontDataCacheKeyHash::hash): Deleted.
(WebCore::FontDataCacheKeyHash::equal): Deleted.
* Source/WebCore/platform/graphics/FontPlatformData.h:
* Source/WebCore/platform/graphics/WidthCache.h:
(WebCore::WidthCache::SmallStringKeyHash::hash): Deleted.
(WebCore::WidthCache::SmallStringKeyHash::equal): Deleted.
* Source/WebCore/rendering/CSSValueKey.h:
(WTF::CSSValueKeyHash::hash): Deleted.
(WTF::CSSValueKeyHash::equal): Deleted.
* Source/WebCore/rendering/TextAutoSizing.h:
(WebCore::TextAutoSizingHash::hash): Deleted.
(WebCore::TextAutoSizingHash::equal): Deleted.
* Source/WebKit/Shared/WebFoundTextRange.h:
(WTF::WebFoundTextRangePDFDataHash::hash): Deleted.
(WTF::WebFoundTextRangePDFDataHash::equal): Deleted.
(WTF::WebFoundTextRangeHash::hash): Deleted.
(WTF::WebFoundTextRangeHash::equal): Deleted.
* Source/WebKit/UIProcess/Extensions/WebExtensionMatchPattern.h:
(WTF::WebExtensionMatchPatternHash::hash): Deleted.
(WTF::WebExtensionMatchPatternHash::equal): Deleted.

Canonical link: https://commits.webkit.org/302472@main
